docs/ape17.rst

```diff
@@ -0,0 +1,533 @@
+``[options]`` section of your ``setup.cfg`` file::
+
+   [options]
+   ...
+   setup_requires = setuptools_scm
+   ...
+
+This is not strictly necessary but will make it possible for ``python setup.py --version``
+to work without having to install ``setuptools_scm`` manually.
+
+Next, check your ``.gitignore`` and make sure that you have a line containing::
+
+   my_package/version.py
+
+Finally, copy over the ``_astropy_init.py`` file generated in :ref:`Step 0 <step0>`, or
+alternatively edit your ``my_package/_astropy_init.py`` file and remove the
+following lines:
+
+.. code:: python
+
+   try:
+       from .version import githash as __githash__
+   except ImportError:
+       __githash__ = ''
+
+and remove ``'__githash__'`` from the ``__all__`` list at the top of the
+file. The git hash is now contained in the version number, so this is no
+longer needed.
+
+Step 9 - Configuring pytest
+---------------------------
+
+To make sure that pytest works properly, you can set a few options in a
+``[tool:pytest]`` section in your ``setup.cfg`` file::
+
+   [tool:pytest]
+   testpaths = "my_package" "docs"
+   astropy_header = true
+   doctest_plus = enabled
+   text_file_format = rst
+   addopts = --doctest-rst
+
+For the ``testpaths`` line, make sure you replace ``my_package`` with
+the name of your package.
+
+The remaining options ensure that the output from pytest includes a
+header that lists dependencies and system information, and also ensure
+that the ``.rst`` files are picked up and tested by pytest.
+
+Step 10 - Update ``MANIFEST.in``
+--------------------------------
+
+Edit your ``MANIFEST.in`` file to remove the following lines, if present
+(and any other line related to ``astropy_helpers``) - those lines might
+include any of the following::
+
+   include ez_setup.py
+   include ah_bootstrap.py
+
+   # the next few stanzas are for astropy_helpers.  It's derived from the
+   # astropy_helpers/MANIFEST.in, but requires additional includes for the actual
+   # package directory and egg-info.
+
+   include astropy_helpers/README.rst
+   include astropy_helpers/CHANGES.rst
+   include astropy_helpers/LICENSE.rst
+   recursive-include astropy_helpers/licenses *
+
+   include astropy_helpers/ez_setup.py
+   include astropy_helpers/ah_bootstrap.py
+
+   recursive-include astropy_helpers/astropy_helpers *.py *.pyx *.c *.h
+   recursive-include astropy_helpers/astropy_helpers.egg-info *
+   # include the sphinx stuff with "*" because there are css/html/rst/etc.
+   recursive-include astropy_helpers/astropy_helpers/sphinx *
+
+   prune astropy_helpers/build
+   prune astropy_helpers/astropy_helpers/tests
+
+Then add a new line near the top with the following::
+
+   include pyproject.toml
+
+Step 11 - Updating your documentation configuration
+---------------------------------------------------
+
+You will need to edit the ``docs/conf.py`` file to make sure it does not
```

Comment:
Also mention the change needed to grab package version number?

Reply:
This doesn't actually need to change compared to before for everything to work - but I want to do a follow up PR cleaning up the ``conf.py`` file so we can maybe discuss what our preferred approach is there?